prover/crypto/mimc/gkrmimc/gkrcircuit.go

```diff
@@ -68,23 +62,37 @@ func createGateNames() {
 		var gateNameBuilder strings.Builder
 		gateNameBuilder.WriteString("mimc")
 		writePaddedHex(&gateNameBuilder, i, nbDigits)
-		gateNames[i] = gateNameBuilder.String()
+		gateNames[i] = gkr.GateName(gateNameBuilder.String())
 	}
 }
 
-// registerGates instantiates and populates the cGkr and gGkr global variables
+// registerGates instantiates and populates the cGkr and gkr global variables
 // which contains the "normal" and the "gnark" version of the GKR gates forming
 // the MiMC GKR circuit.
-func registerGates() {
+func registerGates() error {
 	for i := 4; i < numGates-1; i++ {
-		name := gateNames[i]
-		gGkr.Gates[name] = NewRoundGateGnark(mimc.Constants[i-prefetchSize])
-		cGkr.Gates[name] = NewRoundGateCrypto(mimc.Constants[i-prefetchSize])
+		if err := gkrgates.Register(
+			RoundGate(mimc.Constants[i-prefetchSize]),
+			2,
```

Comment:
cosmetic but helps when re-reading code, can you move the "2" and the "17" magic numbers in a const with a descriptive name just below the call ?

Reply:
Done